src/nanoarrow/testing/testing.cc

```diff
@@ -611,7 +611,7 @@ TestingJSONWriter::TestingJSONWriter()
       include_metadata_(true),
       dictionaries_(new internal::DictionaryContext()) {}
 
-TestingJSONWriter::~TestingJSONWriter() = default;
+TestingJSONWriter::~TestingJSONWriter() { delete dictionaries_; };
```

Comment:
It might be worth adding a quick note about why we have to use new/delete here instead of unique_ptr as I think this will look odd to people who don't develop on Windows often